python/ray/serve/tests/test_logging.py

```diff
@@ -192,6 +202,228 @@ def check_log(
         )
 
 
+@pytest.mark.parametrize("log_format", ["TEXT", "JSON"])
+def test_http_access_log_in_logs_file(serve_instance, log_format):
+    name = "deployment_name"
+    fastapi_app = FastAPI()
+
+    @serve.deployment(name=name)
+    @serve.ingress(fastapi_app)
+    class Handler:
+        def __init__(self):
+            self._replica_unique_id = serve.get_replica_context().replica_id.unique_id
+
+        def _get_context_info(self):
+            """Get context information for matching with logs"""
+            request_context = ray.serve.context._get_serve_request_context()
+            return {
+                "replica": self._replica_unique_id,
+                "request_id": request_context.request_id,
+                "worker_id": ray.get_runtime_context().get_worker_id(),
+                "node_id": ray.get_runtime_context().get_node_id(),
+                "actor_id": ray.get_runtime_context().get_actor_id(),
+            }
+
+        @fastapi_app.get("/")
+        def get_root(self):
+            return self._get_context_info()
+
+        @fastapi_app.post("/")
+        def post_root(self):
+            return self._get_context_info()
+
+        @fastapi_app.get("/{status}")
+        def template(self, status: str):
+            content_info = {"context": self._get_context_info()}
+            return PlainTextResponse(
+                content=json.dumps(content_info),
+                status_code=int(status),
+                media_type="application/json",
+            )
+
+        @fastapi_app.put("/fail")
+        def fail(self):
+            error_response = {"error": "OOPS!", "context": self._get_context_info()}
+            return PlainTextResponse(
+                content=json.dumps(error_response),
+                status_code=500,
+                media_type="application/json",
+            )
+
+    serve.run(Handler.bind(), logging_config={"encoding": log_format})
+
+    # Get log file information
+    client = _get_global_client()
+    serve_log_dir = get_serve_logs_dir()
```

Comment:
you could use this utility function instead https://github.com/ray-project/ray/blob/b2a9f2000248d5a53ccbced4bc6485a81199ef70/python/ray/serve/_private/logging_utils.py#L194